PBM-782 Release notes 1.7.0 (#20)
percona/pbm-docs: source/release-notes/1.7.0.rst

.. _PBM-1.7.0:

================================================================================
*Percona Backup for MongoDB* 1.7.0
================================================================================

:Date:  , 2022
:Installation: `Installing Percona Backup for MongoDB <https://www.percona.com/doc/percona-backup-mongodb/installation.html>`_

Percona Backup for MongoDB is a distributed, low-impact solution for consistent backups of MongoDB
sharded clusters and replica sets. This is a tool for creating consistent backups
across a MongoDB sharded cluster (or a single replica set), and for restoring
those backups to a specific point in time. The project was
inspired by (and intends to replace) the Percona-Lab/mongodb_consistent_backup tool.

Release highlights
======================

New Features
================================================================================

> * :jirabug:`PBM-734`: Add the config option to set debug log levels for S3  requests

Reviewer: ```suggestion
* :jirabug:`PBM-734`: Users can now configure logging levels for S3 requests to debug issues with object storage
```

Can we add the link to the doc about this feature? Or at least a parameter?

Author: Sure. I suggest moving this wording to the release highlights

> * :jirabug:`PBM-742`: Add the ability to :ref:`replay oplog <oplog-replay>` from arbitrary start time

Reviewer: ```suggestion
* :jirabug:`PBM-742`: Add the ability to :ref:`replay oplog <oplog-replay>` from arbitrary start time. This can be used to reduce Recovery Point Objective (RPO) when database is recovered from storage-level backup, for example LVM-snapshots.
```

This should go to highlights as well.

Improvements
================================================================================

> * :jirabug:`PBM-680`: Skip TLS verification for object storage

Reviewer: ```suggestion
* :jirabug:`PBM-680`: Skip TLS verification for object storage. This can be useful for private object storage with self-signed certificates.
```

Can we also add a link to the doc with this flag?

Reviewer: This should also go to highlights. It was highly awaited feature. Users can use private object storages without workarounds.

> * :jirabug:`PBM-770`: Support configurable compression method/level for |PITR| chunks (Thanks to Damiano Albani for reporting this issue)

Reviewer: Can we merge PBM-770 and PBM-756 into one line? One for PITR and another one for backups. 

Something like: 
Support configurable compression method/level for backups and |PITR| chunks (Thanks to Damiano Albani for the contribution)

Also Damiano not only reported this issue, but also contributed the code :)

Author: I suggest merging them in the release highlights section.

> * :jirabug:`PBM-764`: Support Zstandard compression format (Thanks to Damiano Albani for reporting this issue)

Reviewer: ```suggestion
* :jirabug:`PBM-764`: Support Zstandard compression format (Thanks to Damiano Albani for the contribution)
```

> * :jirabug:`PBM-750`: Make max number of S3 upload parts configurable (Thanks to Damiano Albani for reporting this issue)

Reviewer: ```suggestion
* :jirabug:`PBM-750`: Users can configure the number of S3 multipart upload chunks to comply with various S3-compatible storage provider requirements (Thanks to Damiano Albani for the contribution)
```

* :jirabug:`PBM-756`: Support configurable compression level for backups (Thanks to Damiano Albani for reporting this issue)

> * :jirabug:`PBM-577`: Add storage class option for S3

Reviewer: ```suggestion
* :jirabug:`PBM-577`: It is now possible to choose an S3 storage class for granular control over various S3 tiers (Thanks to Damiano Albani for the contribution)
```

Need also to provide a link to the doc with this new option.

Bugs Fixed
================================================================================

> * :jirabug:`PBM-721`: Add the ability to configure retry upload to S3

Reviewer: ```suggestion
* :jirabug:`PBM-721`: Fixed a bug where an upload of the backup to S3-storage was failing due to unstable network connection. |PBM| can now be configured to retry the upload with flexible timeouts.
```

add a link to parameters in docs as well?

.. _PBM-1.7.0:

================================================================================
*Percona Backup for MongoDB* 1.7.0
================================================================================

:Date:  , 2022
:Installation: `Installing Percona Backup for MongoDB <https://www.percona.com/doc/percona-backup-mongodb/installation.html>`_

Percona Backup for MongoDB is a distributed, low-impact solution for consistent backups of MongoDB
sharded clusters and replica sets. This is a tool for creating consistent backups
across a MongoDB sharded cluster (or a single replica set), and for restoring
those backups to a specific point in time. The project was
inspired by (and intends to replace) the Percona-Lab/mongodb_consistent_backup tool.

Release highlights
======================

* Support for physical backups. This drastically speeds up backup performance  for large databases (several terabytes). This is a technical preview feature.
* Oplog replay from the arbitrary start time. This enables users to make a point-in-time recovery from any backup.

> * Configurable debug log levels for S3.

Reviewer: I think we can remove this one from highlights. It does not bring that much value.

.. _PBM-1.7.0:

================================================================================
*Percona Backup for MongoDB* 1.7.0
================================================================================

:Date:  , 2022
:Installation: `Installing Percona Backup for MongoDB <https://www.percona.com/doc/percona-backup-mongodb/installation.html>`_

Percona Backup for MongoDB is a distributed, low-impact solution for consistent backups of MongoDB
sharded clusters and replica sets. This is a tool for creating consistent backups
across a MongoDB sharded cluster (or a single replica set), and for restoring
those backups to a specific point in time. The project was
inspired by (and intends to replace) the Percona-Lab/mongodb_consistent_backup tool.

Release highlights
======================

> * Support for :ref:`physical backups <backup-types>`. This drastically speeds up backup performance  for large databases (several terabytes). This is a technical preview feature.

Author: I think tech preview to be consistent across all products where tech preview features are introduced

sharded clusters and replica sets. This is a tool for creating consistent backups
across a MongoDB sharded cluster (or a single replica set), and for restoring
those backups to a specific point in time. The project was
inspired by (and intends to replace) the Percona-Lab/mongodb_consistent_backup tool.

Release highlights
======================

* Support for :ref:`physical backups <backup-types>` in |PSMDB| starting from versions 4.2.15-16 and 4.4.6-8 and higher. Physical backups drastically speed up backup performance for large databases (several terabytes). This is a technical preview feature. 
* :ref:`Oplog replay <oplog-replay>` from the arbitrary start time. This reduces Recovery Point Objective (RPO) when database is recovered from storage-level backup, for example LVM-snapshots.
* Ability to configure compression method and level for |PITR| chunks and compression level for backups.
* Ability to configure the number of S3 multipart upload chunks to comply with various S3-compatible storage provider requirements. 
* Ability to configure the number of upload retries. This facilitates data upload in case of unstable network connection.
  
Breaking Changes
================

> To use physical backups with filesystem-based backup storage, the `mongod` user must be the owner of the backup directory. 

Reviewer 2: It's not about ownership but about write permissions on filesystem level. Mongod user must have write permissions to backup dir

> To use physical backups with filesystem-based backup storage, the `mongod` user must be the owner of the backup directory. 

Author: Rewritten

> To use physical backups with filesystem-based backup storage, the `mongod` user must be the owner of the backup directory. 

Reviewer 2: Regardless how users install or upgrade pbm (from packages, binaries, manually, whatever) - two conditions must be met:
1. user from whom pbm-agent runs must have read/write permissions on mongo datadir (for physical backup/restore) 
2. user from whom pbm-agent runs must have read/write permissions on backup dir (for filesystem-based storage)
To met this conditions we have changed pbm user in our packages. Starting from this release pbm-agent daemon runs from **mongod** users, not **pbm**. So upgrade procedure from previous releases requires this pre-check
 - If pbm and mongodb was installed from our packages and filesystem storage was in use - users should manually check permissions on backup dir for write access for **mongod** user and change them if needed before upgrade
 - for custom installations - users should check conditions above

.. _PBM-1.7.0:

================================================================================
*Percona Backup for MongoDB* 1.7.0
================================================================================

:Date: April 14, 2022
:Installation: `Installing Percona Backup for MongoDB <https://www.percona.com/doc/percona-backup-mongodb/installation.html>`_

Percona Backup for MongoDB is a distributed, low-impact solution for consistent backups of MongoDB
sharded clusters and replica sets. This is a tool for creating consistent backups
across a MongoDB sharded cluster (or a single replica set), and for restoring
those backups to a specific point in time. The project was
inspired by (and intends to replace) the Percona-Lab/mongodb_consistent_backup tool.

Release highlights
======================

> * Support for :ref:`physical backups <backup-types>` in |PSMDB| starting from versions 4.2.15-16 and 4.4.6-8 and higher. Physical backups drastically speed up backup performance for large databases (several terabytes). This is a technical preview feature. 

Reviewer: ```suggestion
* Support for :ref:`physical backups <backup-types>` in |PSMDB| starting from versions 4.2.15-16 and 4.4.6-8 and higher. Physical backups drastically speed up backup and restore performance for large databases (several terabytes). This is a technical preview feature. 
```

> * :ref:`Oplog replay <oplog-replay>` from the arbitrary start time. This reduces Recovery Point Objective (RPO) when database is recovered from physical backups.

Reviewer: ```suggestion
* :ref:`Oplog replay <oplog-replay>` from the arbitrary start time. This reduces Recovery Point Objective (RPO) when database is recovered from physical or storage-level backups.
```

* Ability to configure compression method and level for |PITR| chunks and compression level for backups.
* Ability to configure the number of S3 multipart upload chunks to comply with various S3-compatible storage provider requirements. 
* Ability to configure the number of upload retries. This facilitates data upload in case of unstable network connection.
  
Breaking Changes
================

To use physical backups with filesystem-based backup storage, the `mongod` user must have the write access to the backup directory. 

New Features
================================================================================

* :jirabug:`PBM-734`: Add the config option to set :ref:`debug log levels <s3-debug-log-levels>` for S3  requests

> * :jirabug:`PBM-805`: Implement physical backups

Reviewer: ```suggestion
* :jirabug:`PBM-805`: Implement physical backups which drastically speed up backup and restore performance for large databases
```
